config/initializers/crossposter_config.rb

```diff
@@ -2,18 +2,18 @@
 
 Rails.application.configure do
   config.x.application_name = ENV.fetch("CROSSPOSTER_APP_NAME", "Mastodon Twitter Crossposter")
-  config.x.domain = ENV.fetch("CROSSPOSTER_DOMAIN")
+  config.x.domain = ENV.fetch("CROSSPOSTER_DOMAIN", nil)
   config.x.repo = ENV.fetch("CROSSPOSTER_REPO", "https://github.com/renatolond/mastodon-twitter-poster")
 
   config.x.stats = ENV.fetch("CROSSPOSTER_STATS", nil)
 
-  config.x.announcement_account_address = ENV.fetch("CROSSPOSTER_FEDI_ACCOUNT_ADDRESS")
-  config.x.announcement_account_at = ENV.fetch("CROSSPOSTER_FEDI_ACCOUNT_AT")
+  config.x.announcement_account_address = ENV.fetch("CROSSPOSTER_FEDI_ACCOUNT_ADDRESS", nil)
+  config.x.announcement_account_at = ENV.fetch("CROSSPOSTER_FEDI_ACCOUNT_AT", nil)
 
   config.x.admin_twitter = ENV.fetch("CROSSPOSTER_ADMIN_TWITTER", nil)
 
-  config.x.admin_fedi_address = ENV.fetch("CROSSPOSTER_ADMIN_FEDI_ADDRESS")
-  config.x.admin_fedi_at = ENV.fetch("CROSSPOSTER_ADMIN_FEDI_AT")
+  config.x.admin_fedi_address = ENV.fetch("CROSSPOSTER_ADMIN_FEDI_ADDRESS", nil)
```

Comment:
I think making those blank will give weird results in the UI, also, I think it's good idea that those are mandatory so that if there's something wrong people can contact someone.

Reply:
The problem is that precompile will crash without those set.

I eventually figured out a work around but it felt gross so setting it to null seemed better.

What about updating the root application controller to check for those and throw up an error. Especially if there's a health check baked into the app (did I include it in this pr? If so I can add it to the docker image too)

Reply:
Can't you do the same as you did with the credentials and set them as placeholder in the docker image?

Reply:
Ah health check is on my local branch. I'll merge it in the morning.

Reply:
I can. I just figured that bit out after I made all the other changes. I can try that instead. It just felt ugly.

I suspect it would have to be done for unit tests (it's been a while but rails probably has a setup script that can set them for testing) but I guess most of the other tasks you'd have everything setup properly anyways.

Okay. Was thinking outloud. I'll include the health check changes since they make kubernetes way nicer and will revert those nil defaults when I wake up